test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -231,8 +262,13 @@ func (sm *StackManager) getStack(ctx context.Context, name string, config runner
 	upCtx, cancel := context.WithTimeout(ctx, stackUpTimeout)
 	var loglevel uint = 1
 	defer cancel()
-	upResult, err := stack.Up(upCtx, optup.ProgressStreams(os.Stderr), optup.DebugLogging(debug.LoggingOptions{
-		LogToStdErr:   true,
+	var logger io.Writer
```

Comment:
I think the code is not correct. The first test that manipulates the stack (so the root `TestXX` that runs the Suite), is going to trigger a `SetLogger`. When an `UpdateEnv` is called in a suite test, you're still going to log in the root test.

Supporting that brings the second question: what happens if you log with a test that is already finished? Which brings the third question: do you want to log the destroy with the last subtest? Probably not. So I'm not sure we need to store the logger, `suite.T()` will still provide the root `Test` in that case.

Reply:
The behaviour in the first version produced clearer logs than before but had a strange behavior.
I applied you advices, I am no longer storing the logger when creating the stack. The logs are now created as follow:
- Logs for initialisation of the stack are created in the first test calling UpdateEnv
- Logs for updates are displayed in the test doing the update
- Logs for destroy are logged in the root test running `suite.Run` since it happens in the `TearDownSuite` function after all test finished

I think the behavior is coherent and each Pulumi logs is in the correct test with that. Let me know if I missed something.
Thanks!